tests/test_permissions_synchronizer.py

```diff
@@ -92,90 +93,64 @@ def reset_test_service(self):
         """
-        resp = utils.test_request(self.url, "GET", "/services/" + self.test_service_name, cookies=self.cookies)
-        if resp.status_code == 200:
-            resp = utils.test_request(self.url, "DELETE", "/services/" + self.test_service_name, cookies=self.cookies)
-            utils.check_response_basic_info(resp, 200, expected_method="DELETE")
-        else:
-            utils.check_response_basic_info(resp, 404, expected_method="GET")
-
-    def create_test_resource(self, resource_name, resource_type, parent_id):
-        # type: (str, str, int) -> int
-        """
-        Creates a test resource in Magpie app.
-        """
-        data = {
-            "resource_name": resource_name,
-            "resource_display_name": resource_name,
-            "resource_type": resource_type,
-            "parent_id": parent_id
-        }
-        resp = utils.test_request(self.url, "POST", "/resources", cookies=self.cookies, json=data)
-        body = utils.check_response_basic_info(resp, 201, expected_method="POST")
-        return body["resource"]["resource_id"]
+        test_magpie.delete_service(self.magpie, self.test_service_name)
 
-    def create_test_permission(self, resource_id, permission, user_name, group_name):
-        # type: (int, Dict, str, str) -> None
+    def create_test_permission(self, resource_id, perm_name, perm_access, perm_scope, user_name, group_name):
+        # type: (int, str, str, str, str, str) -> None
         """
         Creates a test permission in Magpie app.
         """
-        data = {"permission": permission}
         if user_name:
-            resp = utils.test_request(self.url, "POST", f"/users/{user_name}/resources/{resource_id}/permissions",
-                                      cookies=self.cookies, json=data)
-            utils.check_response_basic_info(resp, 201, expected_method="POST")
+            self.magpie.create_permission_by_user_and_res_id(
+                user_name=user_name,
+                res_id=resource_id,
+                perm_name=perm_name,
+                perm_access=perm_access,
+                perm_scope=perm_scope)
         if group_name:
-            resp = utils.test_request(self.url, "POST", f"/groups/{group_name}/resources/{resource_id}/permissions",
-                                      cookies=self.cookies, json=data)
-            utils.check_response_basic_info(resp, 201, expected_method="POST")
+            self.magpie.create_permission_by_grp_and_res_id(
+                grp_name=group_name,
+                res_id=resource_id,
+                perm_name=perm_name,
+                perm_access=perm_access,
+                perm_scope=perm_scope)
 
-    def delete_test_permission(self, resource_id, permission, user_name, group_name):
-        # type: (int, Dict, str, str) -> None
+    def delete_test_permission(self, resource_id, permission_name, user_name, group_name):
+        # type: (int, str, str, str) -> None
```

Comment:
It would be less error-prone to use the enum types as input and resolve `.value` inside the function.